### ISSUE TITLE
Tensor 3D dla sekwencji (Analiza strumienia czasu)

### ISSUE THREAD

**Issue by yabool2001** [yabool2001/BPSK#2]
eśli chcesz, aby model ML przetwarzał surowy ciąg próbek prosto z PlutoSDR (np. samodzielnie uczył się filtracji kształtującej, radził sobie z przesunięciami fazy czy kanałem wielodrogowym), musisz dostarczyć mu wycinek czasu. Wtedy dane $I$ oraz $Q$ traktuje się jako dwa niezależne "kanały" sygnału (podobnie jak lewy i prawy kanał w audio stereo).

Kształt tensora $X$: [wielkość_batcha, 2, długość_sekwencji]

Załóżmy, że do analizy jednego bitu BPSK wycinasz okno składające się z 64 surowych próbek z radia. Twój tensor dla batcha 32 takich okien będzie wyglądał tak:
-   dim=0: 32 (liczba okien w paczce)
- dim=1: 2 (kanały wejściowe: I oraz Q)
- dim=2: 64 (kolejne próbki w czasie)

Taki format (tzw. channel-first) jest idealny i natywnie wymagany, jeśli w swoim rozwiązaniu użyjesz jednowymiarowych sieci konwolucyjnych (nn.Conv1d), które świetnie sprawdzają się w cyfrowym przetwarzaniu sygnałów.

Kiedy pobierzesz dane z PlutoSDR (np. używając biblioteki pyadi-iio), zazwyczaj otrzymasz jednowymiarową tablicę NumPy z liczbami zespolonymi. Aby przekształcić ją w odpowiedni tensor $X$ o kształcie [N, 2], możesz użyć bardzo wygodnej, wbudowanej funkcji PyTorcha:

```python
# Symulacja danych odebranych z PlutoSDR (np. 5 zaszumionych próbek)
pluto_rx_data = np.array([0.8+0.1j, -0.9-0.2j, 0.7-0.1j, -0.8+0.3j, 0.9+0.05j])

# 1. Konwersja z NumPy (complex128) na tensor PyTorch (complex64)
tensor_complex = torch.from_numpy(pluto_rx_data).to(torch.complex64)

# 2. Rozbicie części rzeczywistej (I) i urojonej (Q) na osobne kolumny
X = torch.view_as_real(tensor_complex)

print(X.shape)
# Zwróci: torch.Size([5, 2])

print(X)
# Zwróci:
# tensor([[ 0.8000,  0.1000],
#         [-0.9000, -0.2000],
#         [ 0.7000, -0.1000],
#         [-0.8000,  0.3000],
#         [ 0.9000,  0.0500]])
```
Dzięki torch.view_as_real() unikasz ręcznego indeksowania tablic i od razu uzyskujesz elegancką macierz $X$, gotową do wejścia w pętlę uczącą, zgodnie ze schematem, o którym rozmawialiśmy wcześniej.
